pkg/clustermanager/reconcile_logicalcluster.go

```diff
@@ -85,68 +115,28 @@ func (c *controller) processAddLC(newCluster *lcv1alpha1.LogicalCluster) error {
 		return err
 	}
 
-	// Create cluster
-	var opts pclient.Options
-	//ES: what exactly is this kubeconfig
-	err = provider.Create(c.context, clusterName, opts)
-	if err != nil {
-		logger.Error(err, "failed to create cluster")
-		return err
-	}
-	logger.Info("Done creating cluster", "clusterName", clusterName)
-
-	// Update the new cluster's status
-	newCluster.Status.Phase = lcv1alpha1.LogicalClusterPhaseReady
-	_, err = c.clientset.
-		LogicalclusterV1alpha1().
-		LogicalClusters(GetNamespace(newCluster.Spec.ClusterProviderDescName)).
-		Update(c.context, newCluster, v1.UpdateOptions{})
-	if err != nil {
-		logger.Error(err, "failed to update cluster status.")
-		return err
-	}
-
+	// Async call the provider to create the cluster. Once created, discovery
+	// will set the logical cluster in the Ready state.
+	go provider.Create(c.context, newCluster.Name, pclient.Options{})
 	return err
 }
 
-/***
-func (c *controller) processUpdateLC(ctx context.Context, key any) error {
-	logger := klog.FromContext(ctx)
-	var err error
-	clusterConfig := key.(*lcv1alpha1.LogicalCluster)
-	_, err = c.clientset.
-		LogicalclusterV1alpha1().
-		LogicalClusters(clusterproviderclient.GetNamespace(clusterConfig.Spec.ClusterProviderDescName)).
-		Update(ctx, clusterConfig, v1.UpdateOptions{})
-	if err != nil {
-		logger.Error(err, "failed to update cluster status.")
-		return err
-	}
-	return err
-}
-***/
-
+// processDeleteLC: process an LC object deleted event
+// If the cluster is not managed, then the provider will be asked to delete it.
+// TODO: add a finalizer to the logical cluster object if the cluster is managed.
 func (c *controller) processDeleteLC(delCluster *lcv1alpha1.LogicalCluster) error {
 	logger := klog.FromContext(c.context)
 	var err error
 
 	var opts pclient.Options
-	clusterName := delCluster.Name
-
-	providerInfo, err := c.clientset.LogicalclusterV1alpha1().ClusterProviderDescs().Get(
-		c.context, delCluster.Spec.ClusterProviderDescName, v1.GetOptions{})
-	if err != nil {
-		logger.Error(err, "failed to get provider resource.")
-		return err
-	}
 
-	provider, err := c.GetProvider(providerInfo.Name)
+	provider, err := c.GetProvider(delCluster.Spec.ClusterProviderDescName)
 	if err != nil {
 		logger.Error(err, "failed to get provider client")
 		return err
 	}
 
-	err = provider.Delete(c.context, clusterName, opts)
+	err = provider.Delete(c.context, delCluster.Name, opts)
```

Comment:
Why don't we check if it's managed ?  Are we going to delete unmanaged clusters if the user deletes their LC ?

Reply:
Good point. Based on our prior discussion we need to test and redesign the delete path potentially with a finalizer.